src/mlir/generator.cc

```diff
@@ -246,11 +317,20 @@ namespace mlir::verona
     if (nodes.size() == 1)
     {
       auto name = AST::getID(nodes[0]);
-      if (auto type = typeTable.lookup(name))
+      if (AST::isClassType(ast))
```

Comment:
Sorry, shouldn't this just go inside `generateType`?

Reply:
```C++
auto name = AST::getID(nodes[0]);
return generateType(name);
```
and check `AST::isClassType(ast)` inside `generateType`

Reply:
I know what you mean. That was my first attempt. I didn't do it for some reason that I can't remember. I'll have another go.

Reply:
Ah, yes, so, `generateType` receives a name, not an AST node. This is because we want to be able to fetch/generate types from the compiler implementation as well as from the AST. This is the discussion I was having with Paul above.

My reasoning was either of:
1. Implement the logic in `parseType`, even if slightly redundant, to make the behaviour explicitly misplaced (ie, easy to spot to change later).
2. Pass a second argument to `generateType`, an AST pointer, that defaults to nullptr. If there is one, and it's a class, lower as a class. This looks to me as orthogonal specialisation anti-pattern and not one I'd like to introduce.
3. Create a wrapper `generateType(ast)` that takes the name, checks the table, checks for class type, then returns `generateType(name)` if not. This is slightly better than (2) and probably (1), but creates a new layer of abstraction that may confuse and ultimately lead to misuse of that function.

For the sake of simplicity, I chose (1). It's not the best, but it won't lead to misuse of functions or anti-patterns.

I'd be happy to apply some other solution if you think they're better than the reasoning above.

Reply:
Makes, sense.